primitives/staking/src/lib.rs

```diff
@@ -27,3 +29,53 @@ pub type SessionIndex = u32;
 
 /// Counter for the number of eras that have passed.
 pub type EraIndex = u32;
+
+pub trait PoolsInterface {
+	type AccountId;
+	type Balance;
+
+	fn slash_pool(
+		account_id: &Self::AccountId,
+		slash_amount: Self::Balance,
+		slash_era: EraIndex,
+		active_era: EraIndex,
+	) -> Option<(Self::Balance, BTreeMap<EraIndex, Self::Balance>)>;
+}
+
+/// Trait for communication with the staking pallet.
+pub trait StakingInterface {
+	/// Balance type used by the staking system.
+	type Balance;
+
+	/// AccountId type used by the staking system
+	type AccountId;
+
+	type LookupSource;
```

Comment:
why?

Reply:
So we can pass the targets to the staking `nominate` extrinsic with out any modification. Lmk if there is a different way to do this though, I am not that familiar with the Lookup api